pkg/contorller/contoller.go

```diff
@@ -423,34 +424,63 @@ func (tc *TFExtensionController) deleteQueueUnitInstance(namespace, name string)
 	return nil
 }
 
+func (tc TFExtensionController) getTFJob(namespace, name string) (*tfjobv1.TFJob, error)  {
+	tfJob, err := tc.tfjobInformer.Lister().TFJobs(namespace).Get(name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Warningf("Can not find related tfjob:%v in namespace:%v", name, namespace)
+		}
+		return nil, err
+	}
+	return tfJob, nil
+}
+
 func (tc *TFExtensionController) deleteQueueAnnotationInTFJob(qu *v1alpha1.QueueUnit) error {
 	namespace := qu.Spec.ConsumerRef.Namespace
 	tfJobName := qu.Spec.ConsumerRef.Name
-	tfJob, err := tc.tfjobClient.KubeflowV1().TFJobs(qu.Spec.ConsumerRef.Namespace).Get(context.TODO(), tfJobName, metav1.GetOptions{})
+	tfJob, err := tc.getTFJob(namespace, tfJobName)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			klog.Warningf("Can not find related tfjob:%v for queueunit:%v in namespace:%v", tfJobName, qu.Name, namespace)
-			return err
-		}
 		klog.Errorf("Get tfjob failed %v/%v %v", namespace, tfJobName, err.Error())
 		return err
 	}
-
 	var annotation = map[string]string{}
 	for k, v := range tfJob.Annotations {
 		if k != Suspend {
 			annotation[k] = v
 		}
 	}
 	tfJob.SetAnnotations(annotation)
-
 	// TODO change to patch
 	_, err = tc.tfjobClient.KubeflowV1().TFJobs(namespace).Update(context.TODO(), tfJob, metav1.UpdateOptions{})
```

Comment:
It is better to wrap Pod's update operation in a function

Reply:
agree!

Reply:
agree!